Project_Starter_Files-Building_a_Metrics_Dashboard/reference-app/trial/app.py

```diff
@@ -1,73 +1,93 @@
-from flask import Flask, render_template, request, jsonify
+import logging
+import re
+import requests
 
+
+from flask import Flask, jsonify, render_template
+from flask_opentracing import FlaskTracing
 from jaeger_client import Config
 from jaeger_client.metrics.prometheus import PrometheusMetricsFactory
-from opentelemetry import trace
-from opentelemetry.exporter import jaeger
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
 from opentelemetry.instrumentation.flask import FlaskInstrumentor
 from opentelemetry.instrumentation.requests import RequestsInstrumentor
-from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.sdk.trace.export import (
-    ConsoleSpanExporter,
-    SimpleExportSpanProcessor,
-)
+from prometheus_flask_exporter import PrometheusMetrics
 
-trace.set_tracer_provider(TracerProvider())
-trace.get_tracer_provider().add_span_processor(
-    SimpleExportSpanProcessor(ConsoleSpanExporter())
-)
 
 app = Flask(__name__)
 FlaskInstrumentor().instrument_app(app)
 RequestsInstrumentor().instrument()
 
+metrics = PrometheusMetrics(app)
+# static information as metric
+metrics.info("app_info", "Application info", version="1.0.3")
+
+logging.getLogger("").handlers = []
+logging.basicConfig(format="%(message)s", level=logging.DEBUG)
+logger = logging.getLogger(__name__)
 
-#config = Config(
-#        config={},
-#        service_name='your-app-name',
-#        validate=True,
-#        metrics_factory=PrometheusMetricsFactory(service_name_label='your-app-name')
-#)
-#tracer = config.initialize_tracer()
 
 def init_tracer(service):
-    logging.getLogger('').handlers = []
-    logging.basicConfig(format='%(message)s', level=logging.DEBUG)
 
     config = Config(
         config={
-            'sampler': {
-                'type': 'const',
-                'param': 1,
-            },
-            'logging': True,
+            "sampler": {"type": "const", "param": 1},
+            "logging": True,
+            "reporter_batch_size": 1,
         },
         service_name=service,
+        validate=True,
+        metrics_factory=PrometheusMetricsFactory(service_name_label=service),
     )
 
     # this call also sets opentracing.tracer
     return config.initialize_tracer()
 
-tracer = init_tracer('first-service')
 
-@app.route('/')
+tracer = init_tracer("trial")
+flask_tracer = FlaskTracing(tracer, True, app)
+
+
+@app.route("/")
 def homepage():
     return render_template("main.html")
-    with tracer.start_span('get-python-jobs') as span:
-        homepages = []
-        res = requests.get('https://jobs.github.com/positions.json?description=python')
-        span.set_tag('first-tag', len(res.json()))
+
+
+@app.route("/trace")
+def trace():
```

Comment:
@SudKul this function is decoupled from `homepage` function defined above. Also note that in the `homepage` there's a syntax error; you cannot have logic defined after `return` is called

Reply:
Yeah, you are right. The function execution will not proceed past the `return` statement.